fix for issue #51

dariomt opened the pull request:
This pragma fixes a known problem compiling with VS2013 or VS2015 in Release.
See issue #51 for more info.

dariomt committed:
fix for #51

This pragma fixes a known problem compiling with VS2013 or VS2015 in Release.
See issue #51 for more info.

stevengj commented on cobyla/cobyla.c:
```diff
@@ -1505,6 +1505,12 @@ static nlopt_result trstlp(int *n, int *m, double *a,
+  
+/* This pragma fixes a known problem compiling with VS2013 or VS2015 in Release */
+/* see https://connect.microsoft.com/VisualStudio/feedback/details/1028781/c1001-on-release-build */
+#if defined(_MSC_VER) && (_MSC_VER == 1800 || _MSC_VER == 1900)
```
I would just use `#if defined(_MSC_VER)` here.  Who knows when this will be fixed, and really doesn't hurt to disable this optimization for this particular loops in all VC++ versions.

dariomt commented:
But older versions may produce a warning if they do not recognize the pragma.
We know for sure that this problem started with VS2013 and it hasn't been resolved yet.
Maybe we could leave it as `#if defined(_MSC_VER) && _MSC_VER >= 1800`?

stevengj commented:
Sounds good.

dariomt committed:
The fix fod Visual Studio starts from version 2013 onwards.

This has not been fixed for VS2015, so we keep the macro guard open for future versions of VS.
The vectorization of this loop is not that important anyways.
See #51.

dariomt commented:
Done!

stevengj commented:
Thanks; can you please squash the commits (with `git rebase -i HEAD~2` followed by a forced push) so that I can merge as a single commit?
